File: frontend/src/plugins/simulators/mailbox/manifest.ts
import { lazy } from 'react'
import { Mail } from 'lucide-react'
import type { PluginManifest } from '../../types'

export const manifest: PluginManifest = {
  code: 'mailbox',
  name: 'plugins.mailbox.name',
  description: 'plugins.mailbox.description',
  category: 'simulator',
  icon: Mail,
  defaultColor: 'blue',
  supportedFormats: ['text', 'image'],
  defaultEnabled: true,
  comingSoon: false,
  sortOrder: 4,
  injectTypes: ['mail', 'doc'],
  playerRoute: '/mail',
  PlayerPage: lazy(() => import('./PlayerPage')),
  EventReceiver: lazy(() => import('./EventReceiver')),
  PreviewPopup: lazy(() => import('./PreviewPopup')),
  configSchema: null,
}

File: backend/app/plugins/simulators/mailbox/manifest.py
from app.plugins.base import PluginCategory, PluginManifest

manifest = PluginManifest(
    code="mailbox",
    name="Messagerie",
    name_en="Mailbox",
    description="Simulation de boite mail",
    category=PluginCategory.SIMULATOR,
    icon="Mail",
    default_color="blue",
    supported_formats=["text", "image"],
    default_enabled=True,
    coming_soon=False,
    sort_order=4,
    inject_types=["mail", "doc", "call"],
    router_module="app.routers.webmail",
    router_prefix="/webmail",
)
